metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_connection.py

```diff
@@ -148,6 +152,29 @@ def authenticator_type_is_valid(cls, v, values):
         logger.info(f"using authenticator type '{v}'")
         return v
 
+    @pydantic.validator("token")
+    def validate_token_oauth_config(cls, v, values):
```

Comment:
can we add some unit tests around this config validation - I suspect `_check_oauth_config` might throw errors if you actually try to use this

Reply:
done!